xarray/plot/plot.py

```diff
@@ -151,6 +179,55 @@ def plot(darray, row=None, col=None, col_wrap=None, ax=None, rtol=0.01,
     return plotfunc(darray, **kwargs)
 
 
+def _infer_line_data(darray, x, y, hue):
+    error_msg = ('must be either None or one of ({0:s})'
+                 .format(', '.join([repr(dd) for dd in darray.dims])))
+    ndims = len(darray.dims)
+
+    if x is not None and x not in darray.dims:
+        raise ValueError('x ' + error_msg)
+
+    if y is not None and y not in darray.dims:
+        raise ValueError('y ' + error_msg)
+
+    if x is not None and y is not None:
+        raise ValueError('You cannot specify both x and y kwargs'
+                         'for line plots.')
```

Comment:
It would be good to add tests that trigger each of these error conditions, e.g.,
```
with pytest.raises(ValueError):
    ...  # do something wrong
```